src/tests/ftest/control/dmg_storage_format_perf.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import os
+
+from control_test_base import ControlTestBase
+from ClusterShell.NodeSet import NodeSet
+
+
+class DmgStorageFormatPerfTest(ControlTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test Class Description:
+    Verify storage format performance
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DmgStorageFormatPerfTest object."""
+        super().__init__(*args, **kwargs)
+        self.setup_start_agents = False
+
+    def test_dmg_storage_format_perf(self):
+        """JIRA ID: DAOS-4842
+
+        Test Description: Verify storage format takes less than <max_s_per_device>.
+                          For example: node1 and node2 each have 2 devices.
+                                       Format should take less than 2 * <max_s_per_device>
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium,ib2
+        :avocado: tags=control,dmg
+        :avocado: tags=dmg_storage_format_perf
+        """
+        max_devices = self.get_max_devices()
+        max_s_per_device = self.params.get("max_s_per_device", "/run/dmg_storage_format_perf/*")
+        max_format_time = max_s_per_device * max_devices
+
+        self.log.info("Stopping and erasing system before format")
+        result = self.dmg.system_stop(force=True)
+        if self.dmg.result.exit_status != 0:
+            self.fail("Failed to stop system")
+
+        result = self.dmg.system_erase()
+        if self.dmg.result.exit_status != 0:
+            self.fail("Failed to erase system")
+
+        self.log.info(
+            "Formatting storage with a timeout of {}s per device".format(max_s_per_device))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions